lib/use-realtime-price.tsx

```diff
@@ -0,0 +1,121 @@
+import { gql, SubscriptionResult } from "@apollo/client"
+import * as React from "react"
+import {
+  RealtimePriceWsSubscription,
+  useRealtimePriceWsSubscription,
+} from "../lib/graphql/generated"
+import { useDisplayCurrency } from "../lib/use-display-currency"
+
+gql`
+  subscription realtimePriceWs($currency: DisplayCurrency!) {
+    realtimePrice(input: { currency: $currency }) {
+      errors {
+        message
+      }
+      realtimePrice {
+        timestamp
+        btcSatPrice {
+          base
+          offset
+        }
+        usdCentPrice {
+          base
+          offset
+        }
+        denominatorCurrency
+      }
+    }
+  }
+`
+
+const useRealtimePrice = (
+  currency: string,
+  onSubscriptionDataCallback?: (
+    subscriptionData: SubscriptionResult<RealtimePriceWsSubscription, any>,
+  ) => void,
+) => {
+  const priceRef = React.useRef<number>(0)
+  const { formatCurrency } = useDisplayCurrency()
+  const hasLoaded = React.useRef<boolean>(false)
+
+  const { data } = useRealtimePriceWsSubscription({
+    variables: { currency },
+    onSubscriptionData({ subscriptionData }) {
+      if (onSubscriptionDataCallback) onSubscriptionDataCallback(subscriptionData)
+    },
+  })
+
+  React.useEffect(() => {
+    if (data && !hasLoaded.current) {
+      // Subscription data has loaded for the first time
+      hasLoaded.current = true
```

Comment:
This is nice. It'll help solve the `NaN` issue we do get when the subscription data hasn't loaded.